### ISSUE TITLE
Confused about QUEUE event on timeline

### ISSUE THREAD

**Issue by zarzen** [horovod/horovod#1544]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.3.0
3. Horovod version: 0.18.2
4. MPI version: 4.0.2
5. CUDA version: 10.1
6. NCCL version: 2.4.8
7. Python version:
8. OS and version:
9. GCC version:

**Your question:**
Hi there,
I see the explanation for `QUEUE` at document as following:
```
QUEUE happens when reduction is done with NCCL, and the previous NCCL operation did not finish yet.
```

I am confused here, is that why the current `Allreduce` operation needs to wait for the previous operation?
I found the QUEUE event recorded at [cuda_context_->RecordEvent(event_queue, QUEUE, *stream)](https://github.com/horovod/horovod/blob/master/horovod/common/ops/cuda_operations.cc#L119)
here. 
I am wondering when will the QUEUE end? 
Is this mean every time we do ncclAllreduce we need to wait for the previous ncclAllreduce operation?
I think CUDA can handle multiple ncclAllreduce together right? (I can often observe long QUEUE time, sometimes over 200ms. Reduce the TensorFusion buffer can reduce the QUEUE time, so I thought this QUEUE is the time waiting at horovod-fusion-buffer.)
(confused here, because I completed newbie in CUDA programming)

**Comment by tgaddair** [comment]
Hey @zarzen, in Horovod we limit the number of parallel NCCL streams used at once with the environment variable `HOROVOD_NUM_NCCL_STREAMS` / `horovodrun --num-nccl-streams`.  The default is 1, but you can experiment with increasing it to see if it will improve performance.

In general, I would think that with fusion buffering, you should be taking efficient enough advantage of your network bandwidth that you wouldn't see benefit in increasing it in all cases, but @romerojosh might have some thoughts on when it might be appropriate to increase it.  In #1128, mixed precision training and small local batch sizes were called out, and in their case they saw about a 1 - 2% increase in performance.
